libsolidity/analysis/SyntaxChecker.cpp

```diff
@@ -508,5 +513,31 @@ bool SyntaxChecker::visitNode(ASTNode const& _node)
 		solAssert(m_sourceUnit);
 		solAssert(m_sourceUnit->experimentalSolidity());
 	}
+	auto const* declaration = dynamic_cast<Declaration const*>(&_node);
+	if (declaration)
+		checkFutureKeyword(*declaration);
 	return ASTConstVisitor::visitNode(_node);
 }
+
+
+void SyntaxChecker::checkFutureKeyword(Declaration const& _declaration)
+{
+	std::set<ASTString> const futureKeywords = {
+		"transient",
+		"layout",
+		"at",
+		"error",
+		"super",
+		"this"
+	};
+		if (futureKeywords.count(_declaration.name()))
+		m_errorReporter.warning(
```

Comment:
`if` is indented too much.

```suggestion
	if (futureKeywords.contains(_declaration.name()))
		m_errorReporter.warning(
```